gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+        sample_sum_check(t, subset, dict(group=["adj"], sex=sexes), verbose)
+        sample_sum_check(
+            t,
+            subset,
+            dict(group=["adj"], pop=list(set(pop_names)), sex=sexes),
+            verbose,
+        )
+
+
+def sex_chr_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    info_metrics: List[str],
+    contigs: List[str],
+    verbose: bool,
+) -> None:
+    """
+    Performs sanity checks for annotations on the sex chromosomes.
+    Check:
+        - That metrics for chrY variants in female samples are NA and not 0
+        - That nhomalt counts are equal to female nhomalt counts for all non-PAR chrX variants
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] info_metrics: List of metrics in info struct of input Table.
+    :param List[str] contigs: List of contigs present in input Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    female_metrics = [x for x in info_metrics if "-female" in x or "-XX" in x]
+
+    if "chrY" in contigs:
+        logger.info("Check values of female metrics for Y variants are NA:")
+        t_y = hl.filter_intervals(t, [hl.parse_locus_interval("chrY")])
+        metrics_values = {}
+        for metric in female_metrics:
+            metrics_values[metric] = hl.agg.any(hl.is_defined(t_y.info[metric]))
+        output = t_y.aggregate(hl.struct(**metrics_values))
+        for metric, value in dict(output).items():
+            if value:
+                values_found = t_y.aggregate(
+                    hl.agg.filter(
+                        hl.is_defined(t_y.info[metric]),
+                        hl.agg.take(t_y.info[metric], 1),
+                    )
+                )
+                logger.info(
+                    f"FAILED {metric} = {None} check for Y variants. Values found: {values_found}"
+                )
+            else:
+                logger.info(f"PASSED {metric} = {None} check for Y variants")
+
+    t_x = hl.filter_intervals(t, [hl.parse_locus_interval("chrX")])
+    t_xnonpar = t_x.filter(t_x.locus.in_x_nonpar())
+    n = t_xnonpar.count()
+    logger.info(f"Found {n} X nonpar sites")
+
+    logger.info("Check (nhomalt == nhomalt_female) for X nonpar variants:")
+    female_metrics = [x for x in female_metrics if "nhomalt" in x]
+    for metric in female_metrics:
+        standard_field = metric.replace("-female", "").replace("-XX", "")
+        generic_field_check(
+            t_xnonpar,
+            (t_xnonpar.info[f"{metric}"] != t_xnonpar.info[f"{standard_field}"]),
+            f"{metric} == {standard_field}",
+            [f"info.{metric}", f"info.{standard_field}"],
+            verbose,
+        )
+
+
+def missingness_sanity_checks(
```

Comment:
again, I know I am a broken record, but this is also in the UKBB QC repo, so should it be added to gnomad_methods in a generalized form?